buildSrc/src/main/java/com/google/firebase/gradle/plugins/FirebaseLibraryPlugin.kt

```diff
@@ -0,0 +1,150 @@
+// limitations under the License.
+
+package com.google.firebase.gradle.plugins
+
+import com.android.build.gradle.LibraryExtension
+import com.github.sherter.googlejavaformatgradleplugin.GoogleJavaFormatExtension
+import com.google.common.collect.ImmutableList
+import com.google.common.collect.ImmutableMap
+import com.google.firebase.gradle.plugins.ci.device.FirebaseTestServer
+import com.google.firebase.gradle.plugins.license.LicenseResolverPlugin
+import java.io.File
+import org.gradle.api.JavaVersion
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.attributes.Attribute
+import org.gradle.api.publish.PublishingExtension
+import org.gradle.api.publish.maven.MavenPublication
+import org.gradle.api.publish.tasks.GenerateModuleMetadata
+import org.gradle.kotlin.dsl.getByType
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+class FirebaseLibraryPlugin : Plugin<Project> {
+
+  override fun apply(project: Project) {
+    project.apply(ImmutableMap.of("plugin", "com.android.library"))
+    project.apply(ImmutableMap.of("plugin", LicenseResolverPlugin::class.java))
+    project.apply(ImmutableMap.of("plugin", "com.github.sherter.google-java-format"))
+    project.extensions.getByType(GoogleJavaFormatExtension::class.java).toolVersion = "1.10.0"
+    val firebaseLibrary =
+      project.extensions.create(
+        "firebaseLibrary",
+        FirebaseLibraryExtension::class.java,
+        project,
+        LibraryType.ANDROID
+      )
+    val android = project.extensions.getByType<LibraryExtension>()
+    android.compileOptions {
+      sourceCompatibility = JavaVersion.VERSION_1_8
+      targetCompatibility = JavaVersion.VERSION_1_8
+    }
+
+    // In the case of and android library signing config only affects instrumentation test APK.
+    // We need it signed with default debug credentials in order for FTL to accept the APK.
+    android.buildTypes { getByName("release").signingConfig = getByName("debug").signingConfig }
+
+    android.defaultConfig {
+      buildConfigField("String", "VERSION_NAME", "\"" + project.version + "\"")
+    }
+
+    // see https://github.com/robolectric/robolectric/issues/5456
+    android.testOptions {
+      unitTests.all {
+        it.systemProperty("robolectric.dependency.repo.id", "central")
+        it.systemProperty("robolectric.dependency.repo.url", "https://repo1.maven.org/maven2")
+        it.systemProperty("javax.net.ssl.trustStoreType", "JKS")
+      }
+    }
+
+    setupApiInformationAnalysis(project, android)
+    android.testServer(FirebaseTestServer(project, firebaseLibrary.testLab, android))
+    setupStaticAnalysis(project, firebaseLibrary)
+    configurePublishing(project, firebaseLibrary, android)
+
+    // reduce the likelihood of kotlin module files colliding.
+    project.tasks.withType(KotlinCompile::class.java) {
+      kotlinOptions.freeCompilerArgs = ImmutableList.of("-module-name", kotlinModuleName(project))
```

Comment:
`List` are immutable by default in Kotlin. This would function the identically:
```kotlin
kotlinOptions.freeCompilerArgs = listOf("-module-name", kotlinModuleName(project))
```